File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/IEventOutbox.cs
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;

namespace Volo.Abp.EventBus.Distributed;

public interface IEventOutbox
{
    void EnqueueAsync(OutgoingEventInfo outgoingEvent);

    Task<List<OutgoingEventInfo>> GetWaitingEventsAsync(int maxCount, CancellationToken cancellationToken = default);

    void DeleteAsync(Guid id);

    void DeleteManyAsync(IEnumerable<Guid> ids);
}

File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/IEventOutbox.java
package com.ueueo.eventbus.distributed;

import com.ueueo.ID;
import com.ueueo.threading.CancellationToken;

import java.util.Collection;
import java.util.List;

public interface IEventOutbox
{
    void enqueue(OutgoingEventInfo outgoingEvent);

    List<OutgoingEventInfo> getWaitingEvents(int maxCount, CancellationToken cancellationToken);

    void delete(ID id);

    void deleteMany(Collection<ID> ids);
}
